### PR TITLE
Initial version of development guide

### PR DESCRIPTION
**Pull request by kiranmeduri**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,6 +7,8 @@ GO111MODULE=on
 IMAGE=amazon/app-mesh-controller
 REPO=$(AWS_ACCOUNT).dkr.ecr.$(AWS_REGION).amazonaws.com/$(IMAGE)
 VERSION=v0.3.0
+GIT_COMMIT_SHORT=$(shell git log -1 --format=%h)
+DEV_VERSION=${VERSION}-${GIT_COMMIT_SHORT}
```
Comment (nckturner) [inline Makefile]: I think its better to use to include information about recent tag and dirty working tree:
```
$ git describe --dirty --tags
v0.0.10-22-gcee18f5-dirty
```

I guess the only potential issue is using VERSION vs the most recent git tag, but I think it should be ok referring to the most recent git tag which should be present due to the github release.

File: Makefile [hunk 2]
```diff
@@ -34,19 +36,16 @@ verify-codegen:
...
 	docker build -t $(IMAGE):$(VERSION) .
 
 .PHONY: push
 push:
-ifeq ($(AWS_ACCOUNT),)
-	$(error AWS_ACCOUNT is not set)
-endif
-	docker tag $(IMAGE):latest $(REPO):latest
-	docker push $(REPO):latest
+	docker tag $(IMAGE):$(DEV_VERSION) $(REPO):$(DEV_VERSION)
```
Comment (nckturner) [inline Makefile]: Why remove the AWS_ACCOUNT check?  Its required in by REPO.

File: docs/development.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+```
+make image push
+```
+- Deploy latest changes to appmesh-controller
+```
+make deploy
+```
+- Use examples from [aws-app-mesh-examples](https://github.com/aws/aws-app-mesh-examples/tree/master/walkthroughs) to verify the controller behavior.
+
+## Updating App Mesh CRD
```
Comment (efe-selcuk) [inline docs/development.md]: Should `make go-fmt` also be added here?

File: Makefile (hunk 2 above)
Comment (kiranmeduri) [reply Makefile]: added it back. should we add same check to push-release as it also depends on REPO?

File: Makefile (hunk 1 above)
Comment (kiranmeduri) [reply Makefile]: TIL, thanks for that cmd.

File: docs/development.md (hunk 3 above)
Comment (kiranmeduri) [reply docs/development.md]: added. We should probably make it as pre-commit hooks.

**Pull request merged**